openproblems/data/tabula_muris_senis.py

```diff
@@ -1,62 +1,78 @@
+    if unknown_organs:
+        raise ValueError(
+            f"Unknown organs provided in `organ_list': {', '.join(unknown_organs)}. "
+            f"Known organs are {', '.join(known_organs)}"
+        )
 
-    return subset_df
 
+def matching_dataset(dataset, method_list, organ_list):
+    # if dataset has multiple methods, skip it
+    if len(dataset["assay"]) > 1:
+        return False
 
-def make_anndata_from_filename_and_url(filename, url, test=False):
-    """Takes in filename and url pair. Returns corresponding anndata object."""
-    with tempfile.TemporaryDirectory() as tempdir:
-        filepath = os.path.join(tempdir, filename)
-        scprep.io.download.download_url(url, filepath)
-        adata = sc.read_h5ad(filepath)
-        utils.filter_genes_cells(adata)
+    method = dataset["assay"][0]["label"]
+    method = METHOD_ALIASES[method]
 
-    if test:
-        sc.pp.subsample(adata, n_obs=100)
-        adata = adata[:, :1000]
-        utils.filter_genes_cells(adata)
+    # if organ_list is not empty, we want specific tissues
+    if len(organ_list) > 0 and len(dataset["tissue"]) > 1:
+        return False
 
-    return adata
+    # if organ_list is empty, we want only tissue aggregate objects
+    if len(organ_list) == 0 and len(dataset["tissue"]) < 2:
+        return False
 
+    # if organ_list is not empty, check for specific tissue
+    if len(organ_list) > 0 and dataset["tissue"][0]["label"] not in organ_list:
+        return False
 
-def make_anndata_list(subset_df, test):
-    """Makes anndata from filename/url pair. Adds to list of anndatas.
+    # if method_list is not empty, check for specific method
+    if len(method_list) > 0 and method not in method_list:
+        return False
 
-    Input dataframe that contains filenames and urls to make anndatas from.
-    Returns a list of anndata objects.
-    """
-    adata_list = []
-    for i in range(len(subset_df)):
-        row = subset_df.iloc[i]
-        adata_list.append(
-            make_anndata_from_filename_and_url(row.filename, row.figshare_url)
-        )
-    if test:
-        return adata_list[0]
-    return adata_list
+    return True
+
+
+def load_raw_counts(dataset):
+    dataset_id = dataset["id"]
+    assets_path = (
+        f"/curation/v1/collections/{COLLECTION_ID}/datasets/{dataset_id}/assets"
+    )
+    url = f"{API_BASE}{assets_path}"
+    res = requests.get(url=url)
+    assets = res.json()
+    assets = [asset for asset in assets if asset["filetype"] == "H5AD"]
+    assert len(assets) == 1
+    asset = assets[0]
+
+    filename = f"{COLLECTION_ID}_{dataset_id}_{asset['filename']}"
+    with tempfile.TemporaryDirectory() as tempdir:
+        filepath = os.path.join(tempdir, filename)
+        scprep.io.download.download_url(asset["presigned_url"], filepath)
+        adata = sc.read_h5ad(filepath)
+
+    utils.filter_genes_cells(adata)
+    # If `raw` exists, raw counts are there
+    if hasattr(adata, "raw"):
```

Comment:
This will fail if `raw` is empty. You want `hasattr(adata, "raw") and adata.raw is not None`

```
>>> adata = ad.AnnData(np.array([[1,0],[0,1]]))
>>> hasattr(adata, "raw")
True
>>> adata.raw is None
True
```